src/gui_common/ModalManager.cs

```diff
@@ -272,4 +278,35 @@ private void OnModalLost(TopLevelContainer popup)
 
         modalsDirty = true;
     }
+
+    /// <summary>
+    ///   Called when the parent of a <paramref name="popup"/> is deleted from the scene tree.
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     If a popup is demoted but its parent is invalid, an exception will occur.
+    ///     Deleting the modal here and removing references to it will avoid the exception.
+    ///   </para>
+    /// </remarks>
+    private void OnParentLost(TopLevelContainer popup)
+    {
+        // Remove the original parent since the reference will now be invalid
+        originalParents.Remove(popup);
```

Comment:
I think this should guard against duplicate signals as MakeModal I think can be called multiple times (so the signal registration also happens multiple times).